src/EditorFeatures/Core/FindUsages/ClassifiedSpansAndHighlightSpanFactory.cs

```diff
@@ -44,6 +47,82 @@ internal static class ClassifiedSpansAndHighlightSpanFactory
                 documentSpan.Document, documentSpan.SourceSpan, cancellationToken).ConfigureAwait(false);
         }
 
+        public static async Task<ClassifiedTextElement> BuildClassifiedTextElementForSpansAsync(ImmutableArray<TextSpan> spans, ITextSnapshot snapshot, Document document, CancellationToken cancellationToken)
```

Comment:
I wouldn't mind @CyrusNajmabadi's review of this; I still feel like we have some other code that might already do this?